db/schema_applier.cc

```diff
@@ -748,93 +739,108 @@ static future<> merge_aggregates(distributed<service::storage_proxy>& proxy, con
     });
 }
 
-static future<> do_merge_schema(distributed<service::storage_proxy>& proxy, sharded<db::system_keyspace>& sys_ks, std::vector<mutation> mutations, bool reload)
-{
-    slogger.trace("do_merge_schema: {}", mutations);
+future<schema_persisted_state> schema_applier::get_schema_persisted_state() {
+    schema_persisted_state v;
+    v.keyspaces = co_await read_schema_for_keyspaces(_proxy, KEYSPACES, _keyspaces);
+    v.scylla_keyspaces = co_await read_schema_for_keyspaces(_proxy, SCYLLA_KEYSPACES, _keyspaces);
+    v.tables = co_await read_tables_for_keyspaces(_proxy, _keyspaces, table_kind::table, _affected_tables);
+    v.types = co_await read_schema_for_keyspaces(_proxy, TYPES, _keyspaces);
+    v.views = co_await read_tables_for_keyspaces(_proxy, _keyspaces, table_kind::view, _affected_tables);
+    v.functions = co_await read_schema_for_keyspaces(_proxy, FUNCTIONS, _keyspaces);
+    v.aggregates = co_await read_schema_for_keyspaces(_proxy, AGGREGATES, _keyspaces);
+    v.scylla_aggregates = co_await read_schema_for_keyspaces(_proxy, SCYLLA_AGGREGATES, _keyspaces);
+
+    co_return std::move(v);
+}
+
+future<> schema_applier::prepare(std::vector<mutation>& muts) {
```

Comment:
Is `muts` mutated? Better keep it const if not.

Reply:
Can't be const because maybe_delete_schema_version modifies mutations (this is some legacy workaround)